app/src/androidTest/java/app/wefridge/wefridge/MainActivityTest.kt

```diff
@@ -0,0 +1,67 @@
+package app.wefridge.wefridge
+
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.LargeTest
+import androidx.test.rule.ActivityTestRule
+import androidx.test.runner.AndroidJUnit4
+import org.hamcrest.Matchers.allOf
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class MainActivityTest {
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityTestRule(MainActivity::class.java)
```

Comment:
This is just testing the Login Screen, right? So the activity should be `com.firebase.ui.auth.KickoffActivity`?

won't run on my phone
```
java.lang.IllegalAccessError: Illegal class access: 'androidx.test.core.app.ListFuture' attempting to access 'androidx.concurrent.futures.DirectExecutor' (declaration of 'androidx.test.core.app.ListFuture' appears in /data/app/~~h56LJuGnapIqYwwfbrytGQ==/app.wefridge.wefridge.test-4461bXO3_O8Z-aPOcT0kRw==/base.apk)
	at androidx.test.core.app.ListFuture.directExecutor(ListFuture.java:139)
	at androidx.test.core.app.ListFuture.init(ListFuture.java:101)
	at androidx.test.core.app.ListFuture.<init>(ListFuture.java:84)
	at androidx.test.core.app.DeviceCapture$forceRedrawGlobalWindowViews$1.run(DeviceCapture.kt:151)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:245)
	at android.app.ActivityThread.main(ActivityThread.java:8024)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
```

Reply:
Seems it is an issue with the most recent version https://stackoverflow.com/questions/70258965/android-espresso-test-illegal-class-access-androidx-test-core-app-listfuture, I'll try to change the version

Reply:
@PassiDel @henrijaeger Can you try again?

Reply:
Switching to an older version definitely changed something.
I am getting a `NoMatchingViewException` now:

```
androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: (view.getId() is <2131231038/app.wefridge.wefridge:id/logo> and view.getContentDescription() is "App logo" and view.getParent() (view.getId() is <2131231165> and view.getParent() view.getId() is <16908290>) and (view has effective visibility <VISIBLE> and view.getGlobalVisibleRect() to return non-empty rectangle))

View Hierarchy:
+>DecorView{id=-1, visibility=VISIBLE, width=2000, height=1200, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params={(0,0)(fillxfill) ty=BASE_APPLICATION wanim=0x1030304
  fl=81810100
  pfl=10020000
  fitSides= naviIconColor=0}, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=3} 
|
+->LinearLayout{id=-1, visibility=VISIBLE, width=2000, height=1177, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@f39360, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+-->ViewStub{id=16908721, res-name=action_mode_bar_stub, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@add3019, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+-->FrameLayout{id=-1, visibility=VISIBLE, width=2000, height=1141, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@955dabf, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=36.0, child-count=1} 
|
+--->ActionBarOverlayLayout{id=2131230901, res-name=decor_content_parent, visibility=VISIBLE, width=2000, height=1141, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@c9f14ea, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+---->ContentFrameLayout{id=16908290, res-name=content, visibility=VISIBLE, width=2000, height=1045, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.appcompat.widget.ActionBarOverlayLayout$LayoutParams@3c46e78, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=96.0, child-count=1} 
|
+----->ConstraintLayout{id=2131231165, res-name=root, visibility=VISIBLE, width=2000, height=1045, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@402a98d, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=4} 
|
+------>MaterialProgressBar{id=2131231284, res-name=top_progress_bar, visibility=INVISIBLE, width=2000, height=6, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@4d95742, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+------>AppCompatImageView{id=2131231038, res-name=logo, desc=App-Logo, visibility=VISIBLE, width=108, height=973, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@82c7353, tag=null, root-is-layout-requested=false, has-input-connection=false, x=626.0, y=36.0} 
|
+------>MaterialTextView{id=2131231040, res-name=main_tos_and_pp, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@b429490, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=, input-type=0, ime-target=false, has-links=false} 
|
+------>ScrollView{id=2131230883, res-name=container, visibility=VISIBLE, width=488, height=973, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=true, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@2a9a58e, tag=null, root-is-layout-requested=false, has-input-connection=false, x=1361.0, y=36.0, child-count=1} 
|
+------->LinearLayout{id=2131230843, res-name=btn_holder, visibility=VISIBLE, width=488, height=324, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@c80f1af, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=3} 
|
+-------->SupportVectorDrawablesButton{id=2131230945, res-name=email_button, visibility=VISIBLE, width=476, height=72, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@8e469bc, tag=null, root-is-layout-requested=false, has-input-connection=false, x=6.0, y=18.0, text=Mit einer E-Mail-Adresse anmelden, input-type=0, ime-target=false, has-links=false} 
|
+-------->SupportVectorDrawablesButton{id=-1, visibility=VISIBLE, width=476, height=72, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@4e07d45, tag=null, root-is-layout-requested=false, has-input-connection=false, x=6.0, y=126.0, text=Über Google anmelden, input-type=0, ime-target=false, has-links=false} 
|
+-------->SupportVectorDrawablesButton{id=-1, visibility=VISIBLE, width=476, height=72, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@21e8c9a, tag=null, root-is-layout-requested=false, has-input-connection=false, x=6.0, y=234.0, text=Über GitHub anmelden, input-type=0, ime-target=false, has-links=false} 
|
+---->ActionBarContainer{id=2131230779, res-name=action_bar_container, visibility=VISIBLE, width=2000, height=96, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.appcompat.widget.ActionBarOverlayLayout$LayoutParams@d10a8c1, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+----->Toolbar{id=2131230777, res-name=action_bar, visibility=VISIBLE, width=2000, height=96, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@5f63454, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+------>AppCompatTextView{id=-1, visibility=VISIBLE, width=128, height=41, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.appcompat.widget.Toolbar$LayoutParams@abe0ffd, tag=null, root-is-layout-requested=false, has-input-connection=false, x=36.0, y=27.0, text=WeFridge, input-type=0, ime-target=false, has-links=false} 
|
+------>ActionMenuView{id=-1, visibility=VISIBLE, width=0, height=96, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.appcompat.widget.Toolbar$LayoutParams@67a14f2, tag=null, root-is-layout-requested=false, has-input-connection=false, x=1988.0, y=0.0, child-count=0} 
|
+----->ActionBarContextView{id=2131230785, res-name=action_context_bar, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@988d743, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=0} 
|
+->View{id=16908336, res-name=navigationBarBackground, visibility=VISIBLE, width=2000, height=23, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@1ac41c0, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=1177.0} 
|
+->View{id=16908335, res-name=statusBarBackground, visibility=VISIBLE, width=2000, height=36, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@ce0aef9, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
	at dalvik.system.VMStack.getThreadStackTrace(Native Method)
	at java.lang.Thread.getStackTrace(Thread.java:1736)
	at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:12)
	at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:7)
	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:8)
	at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:12)
	at app.wefridge.wefridge.MainActivityTest.mainActivityTest(MainActivityTest.kt:37)
```